src/game_object.cpp

```diff
@@ -0,0 +1,40 @@
+#include "game_object.h"
+
+GameObject::GameObject(Scene *scene)
+{
+	this->scene = scene;
+}
+
+
+GameObject::~GameObject()
+{
+}
+
+// Attaches a model to the game object
+void GameObject::attach_model(Model *m)
+{
+	this->model = m;
+}
+
+// Renders model in scene
+void GameObject::draw()
+{	
+	if (model)
```

Comment:
Is every `GameObject` guaranteed to have a model? If so, why not just pass it in through the constructor?